File: bots/node/scrapeScores.js
// This script scrapes MLB scores from ESPN and saves them in JSON and CSV formats.
// It uses axios for HTTP requests and cheerio for parsing HTML.
// scrapeScore.js
const axios = require('axios');
const cheerio = require('cheerio');
const fs = require('fs');
const path = require('path');

const today = new Date().toISOString().split('T')[0];
const outputDir = path.join(__dirname, '../../data');

(async () => {
  try {
    const res = await axios.get('https://www.espn.com/mlb/scoreboard');
    const $ = cheerio.load(res.data);
    const games = [];

    $('section.Scoreboard').each((i, el) => {
      const teams = $(el).find('span.sb-team-short').map((i, el) => $(el).text()).get();
      const scores = $(el).find('span.sb-team-score').map((i, el) => $(el).text()).get();
      const status = $(el).find('span.sb-meta').text().trim();

      if (teams.length === 2 && scores.length === 2) {
        games.push({
          date: today,
          home_team: teams[1],
          home_score: scores[1],
          away_team: teams[0],
          away_score: scores[0],
          status: status
        });
      }
    });

    if (!fs.existsSync(outputDir)) fs.mkdirSync(outputDir);

    fs.writeFileSync(path.join(outputDir, `scores_${today}.json`), JSON.stringify(games, null, 2));
    fs.writeFileSync(path.join(outputDir, `scores_${today}.csv`), [
      'date,home_team,home_score,away_team,away_score,status',
      ...games.map(g =>
        `${g.date},"${g.home_team}",${g.home_score},"${g.away_team}",${g.away_score},"${g.status}"`
      )
    ].join('\n'));

    console.log("Scraped and saved scores (Node).");
  } catch (err) {
    console.error("Scraping failed:", err);
  }
})();

File: bots/python/scrape_scores.py
# This script fetches historical MLB player statistics for the seasons 2019 to 2024
# and saves them in JSON and CSV formats. It uses the MLB Stats API to get the data.
# scrape_scores.py - Refactored ESPN scoreboard scraper
import os
import json
import requests
from datetime import datetime

OUTPUT_DIR = "./data"
SCORES_URL = "https://site.api.espn.com/apis/site/v2/sports/baseball/mlb/scoreboard"

def log(msg):
    print(f"[{datetime.now().isoformat()}] {msg}")

def fetch_scores():
    try:
        response = requests.get(SCORES_URL, timeout=10)
        response.raise_for_status()
        return response.json()
    except Exception as e:
        log(f"❌ Failed to fetch scores: {e}")
        return None

def parse_scores(data):
    games = []
    for event in data.get("events", []):
        try:
            competition = event["competitions"][0]
            competitors = competition.get("competitors", [])
            game = {
                "date": event.get("date"),
                "name": event.get("name"),
                "status": competition.get("status", {}).get("type", {}).get("description", ""),
                "homeTeam": "",
                "awayTeam": "",
                "homeScore": "",
                "awayScore": ""
            }

            for team in competitors:
                if team.get("homeAway") == "home":
                    game["homeTeam"] = team.get("team", {}).get("displayName", "")
                    game["homeScore"] = team.get("score", "0")
                elif team.get("homeAway") == "away":
                    game["awayTeam"] = team.get("team", {}).get("displayName", "")
                    game["awayScore"] = team.get("score", "0")

            games.append(game)
        except Exception as e:
            log(f"⚠️ Error parsing event: {e}")
    return games

def save_scores(games, date_str):
    os.makedirs(OUTPUT_DIR, exist_ok=True)
    file_path = os.path.join(OUTPUT_DIR, f"scores_{date_str}.json")
    with open(file_path, "w") as f:
        json.dump(games, f, indent=2)
    log(f"✅ Saved {len(games)} games to {file_path}")

def main():
    log("📡 Fetching today’s MLB scores...")
    today = datetime.now().strftime("%Y-%m-%d")
    raw = fetch_scores()
    if raw:
        parsed = parse_scores(raw)
        save_scores(parsed, today)
    log("🎯 Score scrape complete.")

if __name__ == "__main__":
    main()
